linear.py

```diff
@@ -0,0 +1,34 @@
+class Linear(object):
+	"""Friteman asked me for this class, so I did it"""
+	"""Class which mesures the distance between two points a and b, given in arguments of the class"""
+	"""The arguments of the class MUST be either tuples of the same length or floats """
+	def __init__(self, a, b):
+		self.a = a
+		self.b = b
+
+	def change_a(self, new_a):
+		self.a = new_a
+
+
+	def change_b(self, new_b):
+		self.b = new_b
+
+
+	def compute_dist(self):
+		if type(self.a) == float and type(self.b) == float:
+			return abs(self.a - self.b)
+
+		if type(self.a) == tuple and type(self.b)==tuple:
+			if len(self.a)==len(self.b):
+				res = 0.
+				for i in range(len(self.a)):
+					res += (self.b[i] - self.a[i])**2
```

Comment:
Ici tu peux écrire

```
for (coor_a, coor_b) in zip(self.a, self.b):
    res += (coor_b - coor_a)**2
```

c'est un peu plus python 2020 genre

le pattern `for i in range(len(ma_liste))` ça se voit de moins en moins et c'est considéré comme ce qu'on appelle un "anti-pattern", c'est genre un truc moche et pas efficace en gros (je crois)